spring-integration/src/test/java/com/xing/beetle/spring/RecordingMessageHandler.java

```diff
@@ -0,0 +1,84 @@
+package com.xing.beetle.spring;
+
+import java.util.concurrent.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class RecordingMessageHandler {
+  protected final System.Logger log = System.getLogger(getClass().getName());
+  protected final CopyOnWriteArrayList<String> result = new CopyOnWriteArrayList<>();
+  protected final CopyOnWriteArrayList<String> redelivered = new CopyOnWriteArrayList<>();
+  protected final CopyOnWriteArrayList<String> deadLettered = new CopyOnWriteArrayList<>();
+  protected final CopyOnWriteArrayList<String> queuePolicyMessages = new CopyOnWriteArrayList<>();
+
+  void assertCounts(
+      String messageId, long expResult, long expDeadLetter, long expRedelivered, long timeout) {
+    CompletableFuture<Long> resultCheck =
+        CompletableFuture.supplyAsync(
+            () -> {
+              long c = 0;
```

Comment:
Not a big deal but probably we  can get rid of these repeatations?

Reply:
Refactored the code